### PR TITLE
D8CORE-1200 Prevent home page from being deleted

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Prevent the home page, 404, and 403 nodes from being deleted.

# Need Review By (Date)
- 1/29

# Urgency
- medium

# Steps to Test
1. Checkout this branch.
1. clear caches
1. log in as a site manager
1. Attempt to delete the 404, 403, or home page.
1. validate you aren't able to do so.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
❤️ this but it definitely needs a test.

File: tests/behat/features/Users/Roles.feature [hunk 1]
```diff
@@ -20,31 +20,41 @@ Feature: Roles
...
+    And I should get a 403 HTTP response
 
   Scenario: Check I can log in as the Site Builder role
     Given I am logged in as a user with the "Site Builder" role
     And I am on "admin/content"
     Then I should get a "200" HTTP response
     And I am on "/node/add/stanford_page"
     And I should see "Layout"
+    Then I am on "/node/1/delete"
+    And I should get a 200 HTTP response
```
Comment (jbickar) [inline tests/behat/features/Users/Roles.feature]: 200?

File: tests/behat/features/Users/Roles.feature [hunk 2]
```diff
@@ -20,31 +20,41 @@ Feature: Roles
...
 
   Scenario: Check I can log in as the Site Developer role
     Given I am logged in as a user with the "Site Developer" role
     And I am on "admin/content"
     Then I should get a "200" HTTP response
     And I am on "/node/add/stanford_page"
     And I should see "Layout"
+    Then I am on "/node/1/delete"
+    And I should get a 200 HTTP response
```
Comment (jbickar) [inline tests/behat/features/Users/Roles.feature]: 200?

File: tests/behat/features/Users/Roles.feature (hunk 1 above)
Comment (sherakama) [reply tests/behat/features/Users/Roles.feature]: > Unfortunately this only works for roles without the "Bypass content access control" permission

File: tests/behat/features/Users/Roles.feature (hunk 2 above)
Comment (sherakama) [reply tests/behat/features/Users/Roles.feature]: Developers have all the power.

File: tests/behat/features/Users/Roles.feature [hunk 3]
```diff
@@ -20,31 +20,41 @@ Feature: Roles
     Then I should get a "200" HTTP response
     And I am on "/node/add/stanford_page"
     And I should not see "Layout"
+    Then I am on "/node/1/delete"
```
Comment (sherakama) [inline tests/behat/features/Users/Roles.feature]: This should probably be updated to "Then I am on the front page" and I click "edit" to we are not locked to node/1

**Pull request merged**